File: dev/requirements/behavior/cases/recording-location.6.2.case.swift
import Foundation
import SetlistCore

/// 6.2 — The recording is written as AAC into the app's own container, in a
/// location the user can delete.
let recordingLocation_6_2 = RequirementCase(id: "6.2") {
    let container = FileManager.default.temporaryDirectory
        .appendingPathComponent("setlist-container-\(UUID().uuidString)")
    let session = FakeRecordingSession()
    let coordinator = CaptureCoordinator(session: session, container: container)
    let showID = UUID()

    try coordinator.startShow(showID)
    session.record(90)
    try coordinator.endShow()

    try expectEqual(session.startedURLs.count, 1, "one file was asked for")
    let url = session.startedURLs[0]
    try expectEqual(url.pathExtension, "m4a", "AAC in an MPEG-4 container")
    try expect(
        url.path.hasPrefix(container.path + "/"),
        "the recording was asked for inside the container, at \(url.path)"
    )
    try expect(url.path.contains(showID.uuidString), "the show's audio is filed under the show")
}

File: dev/requirements/behavior/cases/recording-location.6.2.case.dart
import 'package:setlist_core/setlist_core.dart';

import '../../shared/expectations.dart';
import '../../shared/fake_world.dart';
import '../../shared/requirement_case.dart';

/// 6.2 — The recording is written as AAC into the app's own container, in a
/// location the user can delete.
final recordingLocation_6_2 = RequirementCase('6.2', (tester) async {
  final session = FakeRecordingSession();
  final capture = CaptureCoordinator(session: session, containerPath: '/container');

  await capture.startShow('show-1');

  final path = session.started.single;
  expectTrue(
    path.startsWith('/container/'),
    'everything written lives inside the container the user deletes with the app',
  );
  expectTrue(path.endsWith('.m4a'), 'AAC in an MPEG-4 container');
  expectTrue(path.contains('show-1'), 'filed under the show it belongs to');
});
